src/config/mongoose_config_parser_toml.erl

```diff
@@ -620,7 +621,8 @@ service_opt([<<"tracking_id">>, <<"service_mongoose_system_metrics">>|_],  V) ->
 %% path: (host_config[].)modules.*
 -spec process_module(path(), toml_section()) -> [option()].
 process_module([Mod|_] = Path, Opts) ->
-    [{b2a(Mod), parse_section(Path, Opts)}].
+    %% Sort option keys to ensure options could be matched in tests
+    [{b2a(Mod), lists:sort(parse_section(Path, Opts))}].
```

Comment:
I had the same issue in tests now, ended up reusing the equivalence test code:

```
eq_host_config(Result, Config) ->
    [F] = parse(Config), % check for all hosts
    compare_config(Result, F(?HOST)),
    compare_config(Result, parse_host_config(Config)). % Check for a single host

compare_config(C1, C2) ->
    compare_unordered_lists(C1, C2, fun handle_config_option/2).
```

The problem with the version in the reviewed code is that we bend the code to the tests, not otherwise.
If there was another reason for the sorting, I would do it like this - consistently:
But... I am still not convinced about it.

```
parse_section(Path, M) ->
    lists:flatmap(fun({K, V}) ->
                          Key = key(K, Path, V),
                          handle([Key|Path], V)
                  end, lists:sort(maps:to_list(M))).
```

Reply:
We do not preserve original key order in TOML.
The options are already sorted for 95% of cases, it's how maps work, apparently (the only case that is not sorted is the riak key for modules).
We could increase usability in tests by ensuring that they are always sorted.
It also makes analyzing cases with large number of keys easier (i.e. mod_muc module).

Reply:
I just think that the results are proplists and should be tested as such... I wanted to prevent anyone from doing things like `[{opt1, Val1}, {opt2, Val2}] = parse_section(...)` :-)

Reply:
Yeah, I am treating it as orddicts, but only for modules.
The reason is that I am trying to use simplest macros for testing, if a module does not have any tested options.

I could revert the code, but we probably would loose location information in the test errors (which would make fixing tests harder).

I haven't found a consistent way for testing though. I would like to have:
- HOF support - i.e. wrap subtests and execute all subtests in each test. If something fails - report ALL errors, not just first one. Similar to eunit underscore macros. Useful for big tests.
- keep location information, so we can jump to code.
- keep the original code snippet in tests.

Also, I haven't any time to think about it and refactor though.



Though xdd:

```erlang
#{opt1 := Val1 := opt2, Val2} = maps:from_list(parse_section(...))
```

Reply:
Ok, if getting the helper at the top of the stacktrace in the command-line CT report is the problem, you can refactor it according the `parse_section` example above - i would be ok for me.
I still prefer the topmost version though - I am using the HTTP test reports which are not lacking any information - actually the extra frame from the stracktrace is even slightly more informative for me... 

So: your choice, please just do it consistently - not only for modules.